CST-965: added name to boot disk

Eddyredline opened the pull request:
SRE team configured Monitoring Dashboards for bundle and test teams, and in that monitoring there is panel with disk quota usage:

Disk_id is given here, and it’s difficult to figure out, which VM disk belongs too.

We ask to add vm_name_prefix to boot_disk name property in order to fast recognize that disk belong to VM.

The result should be like that:

VM: novoselovdy-adcm-terraform-ubuntu-test

disk_name: novoselovdy-adcm-terraform-ubuntu-test-disk

Before:
![изображение](https://github.com/arenadata/ansible-module-yandex-cloud/assets/18572912/fbb62e20-b114-4069-b070-4f270154a728)

Result:
![Screenshot 2024-03-20 at 14 28 10](https://github.com/arenadata/ansible-module-yandex-cloud/assets/18572912/6f6a6b62-a881-4358-bf7e-c07107ad00b7)

Maunty commented on modules/ycc_vm.py:
```diff
@@ -680,6 +680,7 @@ def _get_instance_params(self, spec):  # pylint: disable=R0914
         metadata = spec.get("metadata")
         labels = spec.get("labels")
         security_groups = spec.get("security_groups")
+        disk_name = hostname + "-disk"  # CST-965: added disk_name to boot disk for Grafana dashboards
```
```suggestion
        disk_name = hostname + "-boot"  # CST-965: added disk_name to boot disk for Grafana dashboards
```

Maunty replied:
IMHO it is more informative.

Eddyredline replied:
Thanks, you are totally wright.
1. Customer asks for -disk pattern in ticket.
2. Do not grafana`s dashboard split disk name on screenshot.
I will ask customer for -boot pattern and change concatenation to f-string

Eddyredline replied:
Fixed. Thanks.

Maunty replied:
Additionally, may be is would be good to save previous behavior for backward compatibility.
Instead of hard-coding this thing, introduce a new parameter.

Maybe other users would like another behavior, the old one, for example.

giggsoff replied:
@Maunty do you mean to add `disk_name` option to `ycc_vm` module (the same we have for `disk_type` and `disk_size`) and use generated one in case of no option defined (assume it is assigned automatically)? It make sense.
_Potentially we can allow to pass `name` into `secondary_disks_spec` and again use generated one in case of no option defined._

In that case caller will set names explicitly. But in that case the name assigning become caller responsibility. We will be forced to modify testing framework logic. Is it ok for us?

Maunty commented:
@Eddyredline What about secondary disks?

Eddyredline commented:
We use JSON format for secondary disk like this:
```
{
                    "autodelete": true,
                    "description": "disk1",
                    "type": "ssd",
                    "size": 10
                },
                {
                    "autodelete": false,
                    "description": "disk2",
                    "type": "hdd",
                    "size": 100
}
```

You can add new key:pair value like **name** to previous pattern. I will notice about this opportunity inside hostrpovider docs.

giggsoff reviewed (commented):
As I understand the same problem exist for additional disks we attach to VM, probably we should also set name for them to identify additional disks on Grafana? For example we can add name prefix into `_get_secondary_disk_specs` function.

giggsoff commented:
> We use JSON format for secondary disk like this:
> 
> ```
> {
>                     "autodelete": true,
>                     "description": "disk1",
>                     "type": "ssd",
>                     "size": 10
>                 },
>                 {
>                     "autodelete": false,
>                     "description": "disk2",
>                     "type": "hdd",
>                     "size": 100
> }
> ```
> 
> You can add new key:pair value like **name** to previous pattern. I will notice about this opportunity inside hostrpovider docs.

As I can see, we cannot. Inside of [_get_secondary_disk_specs](https://github.com/arenadata/ansible-module-yandex-cloud/blob/8a6041f84fba99092bd51811d7e05dc5bcf588e0/modules/ycc_vm.py#L987-L1003) we do not pass `name`.

Eddyredline commented:
Added the disk_name field inside hostprovider`s config to the boot disk.
If the field is explicitly specified, then the boot disk name will be taken from it, if not specified, then the cloud will generate the disk id, as before for backward compatibility.

Added logic for additional disks names for grafana dashboards.
